src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Change the code to avoid repetition, but save readability.
Don’t shorten variables or function names. Check this [guide](https://mate-academy.github.io/style-guides/javascript.html#2311-dont-shorten-variables-or-functions-names).
```suggestion
    const children = people.filter((person) => (
      onlyWithSon 
        ? people.some(mother => mother.name === person.mother) && person.sex === 'm'
        : people.some(mother => mother.name === person.mother)
      )
    )
```